app/controllers/EoriDetailsController.scala

```diff
@@ -31,20 +31,26 @@ class EoriDetailsController @Inject()(cc: ControllerComponents, eoriDetailsServi
 
   def onLoad(id: String): Action[AnyContent] = Action.async { implicit request =>
     eoriDetailsService.retrieveEoriDetails(id).map {
-      case Right(eoriDetails) => Ok(
-        Json.obj(
-          "eori" -> eoriDetails.eori,
-          "name" -> eoriDetails.name,
-          "streetAndNumber" -> eoriDetails.streetAndNumber,
-          "city" -> eoriDetails.city,
-          "postalCode" -> eoriDetails.postalCode,
-          "countryCode" -> eoriDetails.countryCode
+      case Right(eoriDetails) =>
+        val filteredVatId = eoriDetails.vatIds.filter(item => item.countryCode == "GB" && item.vatId.nonEmpty)
+        val vatIdJson = if (filteredVatId.nonEmpty) {
+          Json.obj("vatId" -> filteredVatId.head.vatId)
```

Comment:
I think you are missing a test for when the controller returns a response with the vatId field present.
NB this is not a comment on the code but there wasn't a test file for me to put the comment against! Sorry.